docs/conf.py

```diff
@@ -9,7 +9,9 @@
 
 # django-specific configuration
 from django.conf import settings
-settings.configure()
+
+# Removed for breaking doc building with django-guardian installed.
+# settings.configure()
```

Comment:
then simply remove it, if Sphinx builds without that line.  It has no other purpose.